frontend/packages/console-app/src/components/cloud-shell/redux/reducer.ts

```diff
@@ -0,0 +1,16 @@
+import { Map } from 'immutable';
+import { TerminalAction, Actions } from './action';
+
+export type State = Map<string, any>;
+
+export default (state: State, action: TerminalAction) => {
+  if (!state) {
+    return Map({
+      terminal: { isExpanded: false },
+    });
+  }
+  if (action.type === Actions.TerminalDrawerToggleExpanded)
+    return state.setIn(['terminal', 'isExpanded'], !state.getIn(['terminal', 'isExpanded']));
```

Comment:
we are using `cloud-shell` name everywhere. We should look to rename `terminal` to `cloudShell` here as well.